### PR TITLE
Starting point should be able to handle a Promise returning function

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

**Comment by cdupuis** [comment]
@lievendoclo could you please address the conflicts?

**Comment by lievendoclo** [comment]
@cdupuis done

**Comment by cdupuis** [comment]
Thanks.

File: src/api-helper/command/generator/generatorCommand.ts [hunk 1]
```diff
@@ -220,6 +226,10 @@ async function computeStartingPoint<P extends SeedDrivenGeneratorParameters>(par
     }
 }
 
+export function isProjectPromise(a: any): a is Promise<Project> {
```
Comment (johnsonr) [inline src/api-helper/command/generator/generatorCommand.ts]: I wouldn't export this. The typing is fragile. It returns true on anything that looks like any promise.

Comment (johnsonr) [inline src/api-helper/command/generator/generatorCommand.ts]: I wouldn't export this. The typing is fragile. It returns true on anything that looks like any promise.

File: src/api-helper/command/generator/generatorCommand.ts [hunk 2]
```diff
@@ -207,7 +207,13 @@ async function computeStartingPoint<P extends SeedDrivenGeneratorParameters>(par
         return repoLoader(repoRef);
     } else {
         // It's a function that takes the parameters and returns either a project or a RemoteRepoRef
-        const rr: RemoteRepoRef | Project = (startingPoint as any)(params);
+        const rr: RemoteRepoRef | Project | Promise<Project> = (startingPoint as any)(params);
+        if (isProjectPromise(rr)) {
+            rr.then((p: Project) => {
+                ctx.messageClient.respond(`Using dynamically chosen starting point project \`${p.id.owner}:${p.id.repo}\``);
```
Comment (johnsonr) [inline src/api-helper/command/generator/generatorCommand.ts]: Need to await this as it's a promise

**Pull request merged**